spire/humbug/api.py

```diff
@@ -407,33 +414,119 @@ async def delete_restricted_token_handler(
 
 
 @app.post("/reports", tags=["reports"], response_model=None)
-async def create_reports_handler(
+async def create_report(
     request: Request,
     report: HumbugReport,
+    sync: bool = Query(None),
     db_session: Session = Depends(db.yield_connection_from_env),
 ) -> Response:
     """
-    Add report to integration journal.
+    Add report task to redis cache.
+
+    report task:
+        - report:
+            - **title** (string): Entry title
+            - **content** (string): Entry content
+            - **tags** (list): Entry tags
+        - **bugout_token** (UUID): Humbug token
+    """
+    restricted_token = request.state.token
+    if not sync:
+
+        try:
```

Comment:
There is not timeout here. When connection to redis is not available, this request hangs. This needs to be fixed before we merge to production.